btservice/AndroidBluetoothServer.cpp

```diff
@@ -1,53 +1,214 @@
-/*
-*  This file is part of openauto project.
-*  Copyright (C) 2018 f1x.studio (Michal Szwaj)
-*
-*  openauto is free software: you can redistribute it and/or modify
-*  it under the terms of the GNU General Public License as published by
-*  the Free Software Foundation; either version 3 of the License, or
-*  (at your option) any later version.
-
-*  openauto is distributed in the hope that it will be useful,
-*  but WITHOUT ANY WARRANTY; without even the implied warranty of
-*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-*  GNU General Public License for more details.
-*
-*  You should have received a copy of the GNU General Public License
-*  along with openauto. If not, see <http://www.gnu.org/licenses/>.
-*/
-
 #include "OpenautoLog.hpp"
 #include "btservice/AndroidBluetoothServer.hpp"
+#include <QNetworkInterface>
+#include <QThread>
 
 namespace openauto
 {
 namespace btservice
 {
 
-AndroidBluetoothServer::AndroidBluetoothServer()
+AndroidBluetoothServer::AndroidBluetoothServer(openauto::configuration::IConfiguration::Pointer config_)
     : rfcommServer_(std::make_unique<QBluetoothServer>(QBluetoothServiceInfo::RfcommProtocol, this))
+    , config(std::move(config_))
```

Comment:
these should be flipped right? where the class member has the `_` while the param has none